US-CBP/GTAS-UI
# Feat/flags

fixes #751 

**Reviewers**
This PR is enormous, but the vast majority of the file changes are newly added images, so you can skip those (all in the public folder). 

There's a new dependency and an audit fix, so you must run **npm install or yarn install** after you get latest.

Note that updates to the lookup context and IDB mean that if you switch back to an older frontend branch, you may temporarily get an error referencing an **Unhandled Rejection VersionError**. This can be cleared by going into dev tools and deleting your lookup DB under **Application > Storage > IndexedDB > lookupDB**. It rebuilds itself to the correct version when you refresh with F5.

![IDBDeleteLocation](https://user-images.githubusercontent.com/49160279/118748055-26429600-b829-11eb-949e-a3d27a432814.JPG)



**To Test**
For most of the changes, just verify that the images load, the layout is not obviously broken anywhere, and that app doesn't blow up.

1. The new icons should show in tables beside all references to IATA country and carrier codes (except on the code editor)
2. Small in-table icons should have a click-activated popover zoom of the image except when they are embedded in a navigation link (navigation takes precendence)
3. Sidenav content should trigger a scrollbar when the viewport height is too small
4. Tooltip content should always show in tooltip green except when the content is also a link (links take precedence)
5. The Table components should accurately show the record count when results are filtered in the table header.

Added ~700 country flags and carrier icons (dynamically loaded from the public folder)
Updated the style/layout to accommodate the icons where necessary and clean up various discrepancies
Reduced memory and cpu overhead for lookup data fetches from indexeddb in tooltips
Refactored lookup context code for clarity
New LazyImage component to lazy load images and cache the results to minimize refetches
New CarrierBadge component for lazyloaded carrier image display
Added background image cycling to switch the login and sidenav backgrounds daily
Fixed several layout issues in sidenav when resizing the viewport
Fixed the Table component count bug which did not recalculate when results were filtered in the column filters
